mozilla_aws_cli/listener.py

```diff
@@ -217,18 +221,37 @@ def handle_oidc_redirect_callback():
 @app.route("/shutdown", methods=["GET"])
 def handle_shutdown():
     logger.debug("Shutting down Flask")
-    exit_sigint()
-
-    # this is down to prevent race conditions
+    is_done.set()
+    threading.Timer(2, server.shutdown).start()
```

Comment:
Call shutdown in a couple of seconds.  This could be shorter time, if we know more about how long the shutdown response takes to be delivered.